File: backend/models/book.js
const mongoose = require('mongoose');

const bookSchema = mongoose.Schema({
  title:{type:String,required:true},
  author:{type:String,required:true},
  price:{type:Number,required:true},
  imageURL:{type:String,required:true},
  description:{type:String,required:true},
  creator:{type:mongoose.Schema.Types.ObjectId,ref:'User',required:true}
});

module.exports = mongoose.model('Book',bookSchema);

File: go-backend/models/book.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Book struct {
	Id          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Title       string             `json:"title" validate:"required"`
	Author      string             `json:"author" validate:"required"`
	Price       float64            `json:"price" validate:"required"`
	ImageURL    string             `json:"imageURL" validate:"required"`
	Description string             `json:"description" validate:"required"`
	Creator     primitive.ObjectID `json:"creator" validate:"required"`
}
